.github/workflows/ci.yml

```diff
@@ -0,0 +1,459 @@
+            DocumentRoot PWD
+            <Directory PWD>
+              AllowOverride All
+              Require all granted
+            </Directory>
+            ErrorLog ${APACHE_LOG_DIR}/error.log
+            CustomLog ${APACHE_LOG_DIR}/access.log combined
+          </VirtualHost>
+          EOF
+          cat << EOF > __apache2.php
+          <?php
+          \$pwd = rtrim(trim(shell_exec('pwd')), '/');
+          \$s = file_get_contents('__000-default.conf');
+          \$s = str_replace('PWD', \$pwd, \$s);
+          file_put_contents('__000-default.conf', \$s);
+          EOF
+          php __apache2.php
+          sudo cp __000-default.conf /etc/apache2/sites-enabled/000-default.conf
+          rm __000-default.conf
+          sudo a2enmod rewrite
+          # run apache as 'runner:docker' instead of 'www-data:www-data'
+          sudo sh -c "echo 'export APACHE_RUN_USER=runner' >> /etc/apache2/envvars"
+          sudo sh -c "echo 'export APACHE_RUN_GROUP=docker' >> /etc/apache2/envvars"
+          sudo systemctl restart apache2
+          echo "Apache has been configured"
+
+      # This is shared between runs, not just jobs. It means the first time the repo runs the job it'll
+      # need to download requirements for the first time, after that it will be plenty quick
+      # https://docs.github.com/en/actions/advanced-guides/caching-dependencies-to-speed-up-workflows
+      - name: Enable shared composer cache
+        uses: actions/cache@937d24475381cd9c75ae6db12cb4e79714b926ed # @v2
+        with:
+          path: ~/.cache/composer
+          key: shared-composer-cache
+
+      - name: Composer
+        env:
+          INPUTS_COMPOSER_REQUIRE_EXTRA: ${{ inputs.composer_require_extra }}
+          MATRIX_COMPOSER_REQUIRE_EXTRA: ${{ matrix.composer_require_extra }}
+        run: |
+          # Remove double quotes
+          INPUTS_COMPOSER_REQUIRE_EXTRA=${INPUTS_COMPOSER_REQUIRE_EXTRA//\"/}
+          MATRIX_COMPOSER_REQUIRE_EXTRA=${MATRIX_COMPOSER_REQUIRE_EXTRA//\"/}
+
+          # Update composer.json and install dependencies
+          # github.base_ref is only available on pull-requests events is the target branch - is is NOT prefixed with refs/heads/
+          # github.ref_name is used for regular branch builds on events push - it is NOT prefixed with refs/heads/
+          # github.ref_name is also the tag on tag events - it is NOT prefixed with refs/tags/
+          # https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability
+
+          # Note: $BRANCH will be tag for tag events
+          BRANCH=$(php -r "echo preg_replace('#^pulls/([0-9\.]+)/.+\$#', '\$1', '${{ github.base_ref }}'?:'${{ github.ref_name }}');")
+          if [[ "$BRANCH" =~ ^[1-9]$ ]] || [[ "$BRANCH" =~ ^[1-9]\.[0-9]+$ ]]; then export COMPOSER_ROOT_VERSION="${BRANCH}.x-dev"; elif [[ "$BRANCH" =~ ^[1-9]\.[0-9]+\.[0-9]+ ]]; then export COMPOSER_ROOT_VERSION="${BRANCH}"; else export COMPOSER_ROOT_VERSION="dev-${BRANCH}"; fi
```

Comment:
Updated so no longer on a single line

Have updated regex to support 0.x versions

https://getcomposer.org/doc/03-cli.md#composer-root-version

We used because we include installer which itself includes recipes.  COMPOSER_ROOT_VERSION is required when testing core modules.  Let's say the current module is silverstripe/assets, one of the required recipe composer.json will define say `"silverstripe/assets": "^1.8", however the root composer.json doesn't define a version.  COMPOSER_ROOT_VERSION will define that version.